enderio-base/src/main/java/com/enderio/base/data/recipe/standard/MaterialRecipes.java

```diff
@@ -29,5 +41,231 @@ protected void buildCraftingRecipes(@Nonnull Consumer<FinishedRecipe> recipeCons
+            .pattern("S S")
+            .pattern(" S ")
+            .define('S', Tags.Items.RODS_WOODEN)
+            .unlockedBy("has_ingredient", InventoryChangeTrigger.TriggerInstance.hasItems(ItemPredicate.Builder.item().of(Tags.Items.RODS_WOODEN).build()))
+            .save(recipeConsumer);
+        
+        ShapedRecipeBuilder.shaped(EIOItems.GEAR_WOOD.get())
+            .pattern("S S")
+            .pattern("   ")
+            .pattern("S S")
+            .define('S', Tags.Items.RODS_WOODEN)
+            .unlockedBy("has_ingredient", InventoryChangeTrigger.TriggerInstance.hasItems(ItemPredicate.Builder.item().of(Tags.Items.RODS_WOODEN).build()))
+            .save(recipeConsumer, new ResourceLocation(EnderIO.MODID, EIOItems.GEAR_WOOD.get().getRegistryName().getPath() +"_corner"));
+        
+        ShapedRecipeBuilder.shaped(EIOItems.GEAR_STONE.get())
+            .pattern("NIN")
+            .pattern("I I")
+            .pattern("NIN")
+            .define('N', Tags.Items.RODS_WOODEN)
+            .define('I', Tags.Items.COBBLESTONE)
+            .unlockedBy("has_ingredient", InventoryChangeTrigger.TriggerInstance.hasItems(ItemPredicate.Builder.item().of(Tags.Items.COBBLESTONE).build()))
+            .save(recipeConsumer);
+        
+        ShapedRecipeBuilder.shaped(EIOItems.GEAR_STONE.get())
+            .pattern(" I ")
+            .pattern("IGI")
+            .pattern(" I ")
+            .define('I', Tags.Items.COBBLESTONE)
+            .define('G', EIOItems.GEAR_WOOD.get())
+            .unlockedBy("has_ingredient", InventoryChangeTrigger.TriggerInstance.hasItems(EIOItems.GEAR_WOOD.get()))
+            .save(recipeConsumer, new ResourceLocation(EnderIO.MODID, EIOItems.GEAR_STONE.get().getRegistryName().getPath() + "_upgrade"));
+        
+        upgradeGear(recipeConsumer, EIOItems.GEAR_IRON.get(), EIOItems.GRAINS_OF_INFINITY.get(), Tags.Items.INGOTS_IRON, Tags.Items.NUGGETS_IRON);
+        upgradeGear(recipeConsumer, EIOItems.GEAR_ENERGIZED.get(), EIOItems.GEAR_IRON.get(), EIOItems.ENERGETIC_ALLOY_INGOT.get(), EIOItems.ENERGETIC_ALLOY_NUGGET.get());
+        upgradeGear(recipeConsumer, EIOItems.GEAR_VIBRANT.get(), EIOItems.GEAR_ENERGIZED.get(), EIOItems.VIBRANT_ALLOY_INGOT.get(), EIOItems.VIBRANT_ALLOY_NUGGET.get());
+        upgradeGear(recipeConsumer, EIOItems.GEAR_DARK_STEEL.get(), EIOItems.GEAR_ENERGIZED.get(), EIOItems.DARK_STEEL_INGOT.get(), EIOItems.DARK_STEEL_NUGGET.get());
+
+        ShapedRecipeBuilder.shaped(EIOItems.WEATHER_CRYSTAL.get())
+            .pattern(" P ")
+            .pattern("VEV")
+            .pattern(" P ")
+            .define('P', EIOItems.PULSATING_CRYSTAL.get())
+            .define('V', EIOItems.VIBRANT_CRYSTAL.get())
+            .define('E', EIOItems.ENDER_CRYSTAL.get())
+            .unlockedBy("has_ingredient", InventoryChangeTrigger.TriggerInstance.hasItems(EIOItems.PULSATING_CRYSTAL.get()))
+            .save(recipeConsumer);
+        
+        // TODO Own crafting type return bucket?
```

Comment:
I think it will give the bucket back; try it out in survival and see :)

Reply:
Yeah it does give you the bucket.